packages/framework/ai-collab/src/test/explicit-strategy/utils.ts

```diff
@@ -76,3 +78,101 @@ export function initializeOpenAIClient(service: "openai" | "azure"): OpenAI {
 		return client;
 	}
 }
+
+/**
+ * A utility class for snapshot testing.
+ */
+export class MochaSnapshotUnitTester {
+	public constructor(
+		public readonly snapshotDirectory: string,
+		public readonly suiteName: string,
+	) {}
+
+	public expectToMatchSnapshot(
+		mochaContext: Mocha.Context,
+		output: string,
+		options?: { fileNameOverride?: string; metadata?: Record<string, string> },
+	): void {
+		const snapshotBaseFolderDir = path.join(this.snapshotDirectory, "__snapshots__");
+
+		// Create the __snapshots__ directory if it does not exist
+		if (!fs.existsSync(snapshotBaseFolderDir)) {
+			fs.mkdirSync(snapshotBaseFolderDir);
+		}
+
+		// Directory to store snapshots
+		const snapshotDir: string = path.join(
+			this.snapshotDirectory,
+			"__snapshots__",
+			this.suiteName,
+		);
+
+		if (!fs.existsSync(snapshotDir)) {
+			fs.mkdirSync(snapshotDir);
+		}
+
+		const testName: string =
+			options?.fileNameOverride ??
+			mochaContext.test?.title.replace(/\s+/g, "_") ??
+			"unknown_test";
+		const snapshotFile: string = path.join(snapshotDir, `${testName}.snap`);
+
+		const shouldUpdateSnapshot: boolean = process.env.UPDATE_SNAPSHOTS === "true";
+
+		if (fs.existsSync(snapshotFile) && !shouldUpdateSnapshot) {
+			// Snapshot exists, compare outputs
+			const fileContent: string = fs.readFileSync(snapshotFile, "utf8");
+			const expectedOutput = this.removeMetadata(fileContent);
+			assert.strictEqual(
+				output.trim(),
+				expectedOutput.trim(),
+				`Snapshot mismatch for test: ${testName}`,
+			);
+		} else {
+			// Add metadata headers
+			const metadataJson: Record<string, string> = { ...options?.metadata };
+			if (mochaContext.test?.parent?.title !== undefined) {
+				metadataJson["Test Suite Title"] = mochaContext.test?.parent?.title;
+			}
+			if (mochaContext.test?.title !== undefined) {
+				metadataJson["Test Title"] = mochaContext.test?.title;
+			}
+			const metadata = this.generateMetadata(metadataJson);
+
+			const snapshotContent = `${metadata}\n\n${output.trim()}`;
+
+			// Save the snapshot
+			fs.writeFileSync(snapshotFile, snapshotContent, "utf8");
+			console.log(
+				`Snapshot ${fs.existsSync(snapshotFile) ? "updated" : "created"} for test: ${testName}`,
```

Comment:
At this point the file will always exist no matter what, so I think we can just go with this.
```suggestion
				`Snapshot updated for test '${testName}'`,
```

Reply:
It will always exist, but its existence was via either it being newly created or update, so this conditional should still make sense.

Reply:
I don't think so... Since the check is an `fs.existsSync()` call (not some other variable that says if the file existed before or not), the message will _always_ say "updated" even if the file didn't exist before, because at this point we already wrote it.

Reply:
Ah yes you're right, this has been updated to include an outer scoped variable from the if/else block `doesFileAlreadyExist` that checks if the file exists prior it to being anything saved -- so, it should make sense now. (Updated in commit 6fc87d9de1a64f7fb50d2294ce8a29b8ced9e904)